lemminx-maven/src/test/java/org/eclipse/lemminx/extensions/maven/MavenProjectCacheTest.java

```diff
@@ -76,6 +88,63 @@ public void testParentChangeReflectedToChild()
 		}
 	}
 
+	@Test
+	public void testAddFolders_didChangeWorkspaceFolders() throws Exception {
+		XMLLanguageService languageService = new XMLLanguageService();
```

Comment:
I suggest you that you:

 * test range of diagnostics
 * use XMLAssert#testDiagnosticsFor from LemMinx

See https://github.com/eclipse/lemminx/blob/d27272d14db97cba7f22a22d7a7c7bc104d28d14/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSyntaxDiagnosticsTest.java#L48 for instance.

Reply:
We're verifying that NO Error diagnostics appear on the document, while, as far as I understand, `testDiagnosticsFor` always verify presence of a certain Error diagnostics code - so it doesn't applicable for this case.

Or I just couldn't find the correct way to use it...
Possible error in our case could be:
```
Diagnostic [
  range = Range [
    start = Position [
      line = 8
      character = 10
    ]
    end = Position [
      line = 8
      character = 11
    ]
  ]
  severity = Error
  code = null
  source = null
  message = "Non-resolvable parent POM for org.test.modules:ModuleC:0.0.1-SNAPSHOT: Could not find artifact org.test.modules:ModuleA:pom:0.0.1-SNAPSHOT"
  tags = null
  relatedInformation = null
]
```
So, if I replace the existing `assertFalse` to `testDiagnosticsFor`, ie do the following change:
```
//		assertFalse(diagnostics.stream().anyMatch(diag -> (diag.getMessage().contains("ModuleA"))));
		
		URI uri = getClass().getResource("/modules/dependent/module-c-pom.xml").toURI();
		testDiagnosticsFor(doc.getTextDocument().getText(), null, null, uri.toString(),  d(8, 10, 8, 11, null));
``` 
then `testDiagnosticsFor` call doesn't detect the failure if it appears - it's just doesn't work for this case